### PR TITLE
use inferface instead class or type class in docs

### PR DESCRIPTION
**Pull request by zjhmale**
relates to #3375

### CODE REVIEW

File: src/Idris/Core/TT.hs [hunk 1]
```diff
@@ -155,7 +155,7 @@ fcIn (FC fn1 (sl1, sc1) (el1, ec1)) (FC fn2 (sl2, sc2) (el2, ec2)) =
   (sl1 == sl2 && sc1 > sc2 || sl1 > sl2) &&
   (el1 == el2 && ec1 < ec2 || el1 < el2)
 
--- | Ignore source location equality (so deriving classes do not compare FCs)
+-- | Ignore source location equality (so deriving interfaces do not compare FCs)
```
Comment (Melvar) [inline src/Idris/Core/TT.hs]: Since this is talking about Haskell code, the replacement is incorrect.

**Comment by Melvar** [comment]
I see a few minor problems due to this being an unchecked search-and-replace. One instance of “type class” split over lines thus becoming “type interface”, several instances of “a (type) class” becoming “a interface” instead of “an interface”.

**Comment by zjhmale** [comment]
all checked 😃 
>I see a few minor problems due to this being an unchecked search-and-replace. One instance of “type class” split over lines thus becoming “type interface”, several instances of “a (type) class” becoming “a interface” instead of “an interface”.

**Comment by zjhmale** [comment]
@ahmadsalim @melted @Melvar please review.

File: libs/oldeffects/Effects.idr [hunk 2]
```diff
@@ -235,7 +235,7 @@ eff {xs = [l ::: x]} env (l :- prog) k
    = let env' = unlabel env in
          eff env' prog (\p', envk => k p' (relabel l envk))
 
--- yuck :) Haven't got type class instances working nicely in tactic
+-- yuck :) Haven't got interface instances working nicely in tactic
```
Comment (ahmadsalim) [inline libs/oldeffects/Effects.idr]: Should be `interface implementations`

File: src/Idris/AbsSyntax.hs [hunk 3]
```diff
@@ -434,7 +434,7 @@ pop_estack = do i <- getIState
     where ptail [] = []
           ptail (x : xs) = xs
 
--- | Add a class instance function.
+-- | Add an interface instance function.
```
Comment (ahmadsalim) [inline src/Idris/AbsSyntax.hs]: Again, "interface implementation". I would try to check whether this happens other places as well.

**Comment by ahmadsalim** [comment]
@zjhmale I added some comments, but otherwise it looks mostly fine.

File: src/Idris/AbsSyntax.hs (hunk 3 above)
Comment (zjhmale) [inline src/Idris/AbsSyntax.hs]: right, we now use `implementation` keywords to implement interface. i'll check this out.

Comment (zjhmale) [inline src/Idris/AbsSyntax.hs]: so if change `interface instance` to `interface implementation`, should we change all the `instance` to `implementation`?

**Comment by zjhmale** [comment]
@ahmadsalim so if now change `interface instance` to `interface implementation`, should we change all the `instance` to `implementation` in the rest of the code base?

**Comment by ahmadsalim** [comment]
@zjhmale Yes, that would be useful to do I think.

**Comment by zjhmale** [comment]
i'll work on it.
>@zjhmale Yes, that would be useful to do I think.

**Comment by zjhmale** [comment]
poke @ahmadsalim @Melvar

**Comment by ahmadsalim** [comment]
It looks good to me.

**Comment by ahmadsalim** [comment]
Thanks for the PR!

**Pull request merged**